yt/frontends/boxlib/data_structures.py

```diff
@@ -1650,11 +1650,11 @@ def _parse_parameter_file(self):
                     self.parameters[l[0].strip()] = l[1].strip()
 
         # set the periodicity based on the integer BC runtime parameters
-        is_periodic = self.parameters["geometry.is_periodic"].split()
-        periodicity = [bool(val) for val in is_periodic]
-        for _ in range(self.dimensionality, 3):
-            periodicity += [True]  # pad to 3D
-        self.periodicity = ensure_tuple(periodicity)
+        self.periodicity[:] = False
+        if "geometry.is_periodic" in self.parameters:
+            is_periodic = self.parameters["geometry.is_periodic"].split()
+        self.periodicity[:len(is_periodic)] = [val == "1" for val in is_periodic]
+        self.periodicity = ensure_tuple(self.periodicity)
```

Comment:
btw, I know it's not your fault, but this line is not only useless but also harmful since it overwrites the original type for this attribute (`MuttableAttribute`).
Sorry you happened to touch such a buggy bunch of lines !